### PR TITLE
[Base] Implemented API Testing Screen with dynamic arguments - resolves #88

### LINKED ISSUE #88: [Base] - Add API Test Screen
- [x] Issue does not already exist
- Identify exactly what aspect your issue is about
  - [x] ignite base application
  - [ ] ignite the cli
  - [ ] ignite generated code

Need for the API test screen has been stubbed.  Please add it.

### PR DESCRIPTION
**Pull request by kevinvangelder**
#### Ignite Base Change
- [X] Works on iOS/Android/XDE
- [X] Tests Pass (run: `npm run test`)
- [X] Code is StandardJS compliant (run: `npm run lint`)

### CODE REVIEW

**Comment by kevinvangelder** [comment]
Haven't been able to test this in Android yet because it shows an App Preferences screen?

File: ignite-base/App/Containers/APITestingScreen.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      // this.props.navigator.setState({title})
+      this.refs.result.setState({message: FJSON.plain(response.data), title: title})
+    } else {
+      this.refs.result.setState({message: response.problem, title: title})
+    }
+  }
+
+  tryEndpoint (apiEndpoint) {
+    const { label, endpoint, args } = apiEndpoint
+    this.api[endpoint].apply(this, args).then((result) => {
```
Comment (kevinvangelder) [inline ignite-base/App/Containers/APITestingScreen.js]: *MAGIC*

File: ignite-base/App/Containers/APITestingScreen.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    const { label, endpoint, args } = apiEndpoint
+    return (
+      <FullButton text={label || `${endpoint}(${args.join(', ')})`} onPress={this.tryEndpoint.bind(this, apiEndpoint)} styles={{marginTop: 10}} key={`${endpoint}-${args.join('-')}`} />
+    )
+  }
+
+  renderButtons () {
+    const endpoints = [
+      { label: 'Get City (Boise)', endpoint: 'getCity', args: ['Boise'] },
+      { endpoint: 'getCity', args: ['Toronto', 'Fail'] }
```
Comment (kevinvangelder) [inline ignite-base/App/Containers/APITestingScreen.js]: `label` is optional, `args` are passed in the order they are given, and over-arity args are ignored.

**Comment by GantMan** [comment]
👍  on this @kevinvangelder

**Pull request merged**